### PR TITLE
[SYCL][sycl-post-link] Add check for deleted Spec Constant Operands

### PR DESCRIPTION
**Pull request by asudarsa**
When we turn on -emit-only-kernel-as-entry-points by default for sycl-post-link tools, there are some entries in some of the specialization constant metadata are not carried over. Hence, a check is required to ensure we are not parsing 'null' specialization constants.

Thanks

### CODE REVIEW

**Comment by asudarsa** [comment]
I will add a test soon.

File: llvm/tools/sycl-post-link/SpecConstants.cpp [hunk 1]
```diff
@@ -1004,9 +1004,12 @@ bool SpecConstantsPass::collectSpecConstantDefaultValuesMetadata(
 
   unsigned Offset = 0;
   for (const auto *Node : N->operands()) {
-    auto *Constant = cast<ConstantAsMetadata>(Node->getOperand(0))->getValue();
-    collectCompositeElementsDefaultValuesRecursive(M, Constant, Offset,
-                                                   DefaultValues);
+    if (Node->getOperand(0)) {
+      auto *Constant =
+          cast<ConstantAsMetadata>(Node->getOperand(0))->getValue();
```
Comment (maksimsab) [inline llvm/tools/sycl-post-link/SpecConstants.cpp:1009]:
```cpp
if (auto *O = Node->getOperand(0); O) {
      auto *Constant =
          cast<ConstantAsMetadata>(O)->getValue();
```

**Review by maksimsab — COMMENTED** [review]
Looking forward for a test.

**Comment by maarquitos14** [comment]
@asudarsa is this still of interest?

**Pull request closed without merging**